File: src/Cpp/10_Listy_dwie_listy/Zad9.cpp
#include <algorithm>
#include <cassert>
#include <vector>

//Otrzymujesz dwie listy. Usun z pierwszej listy te
//elementy, ktore wystepuja rowniez w drugiej liscie.

//Zlozonosc obliczeniowa O(n^2)
//Zlozonosc pamieciowa O(n)
void usunCzescWspolnaV1(std::vector<int>& listaA, std::vector<int>& listaB)
{

    auto it = listaA.begin();

    while (it != listaA.end()) {
        if (find(listaB.begin(), listaB.end(), *it) != listaB.end())
            it = listaA.erase(it);
        else
            it++;
    }
}

void test1()
{
    std::vector<int> listaA{ 3, 6, 2, 7, 9 };
    std::vector<int> listaB{ 4, 2, 3, 5, 6 };
    std::vector<int> wynik{ 7, 9 };

    usunCzescWspolnaV1(listaA, listaB);

    assert(listaA == wynik);
}

int main()
{

    test1();

    return 0;
}

File: src/JavaScript/10_Listy_dwie_listy/Zad9.js
    usunCzescWspolnaV1 = function(listaA, listaB) {
        for (var i = 0; i < listaA.length; i++) {
            {
                if ((listaB.indexOf((listaA[i])) >= 0)) {
                    listaA.splice(i, 1)[0];
                    i--;
                }
            };
        }
        return listaA;
    };
    test1 = function() {
        var listaA = [3, 6, 2, 7, 9]
        var listaB = [4, 2, 3, 5, 6]
        var oczekiwane = [7, 9]
        var wynik = usunCzescWspolnaV1(listaA, listaB);
        if (wynik.length !== oczekiwane.length) {
            throw new Error(`Assertion error line 29: ${wynik.length} != ${oczekiwane.length}`);
        }
        for (var i = 0; i < wynik.length; i++) {
            if (wynik[i] !== oczekiwane[i]) {
                throw new Error(`Assertion error line 33: ${wynik[i]} != ${oczekiwane[i]}`);
            }
        }
    };
    main = function(args) {
        test1();
    };

    main(null);